src/polyglot-notebooks-vscode-common/src/commands.ts

```diff
@@ -179,34 +179,37 @@ export function registerFileCommands(context: vscode.ExtensionContext, parserSer
         'Jupyter Notebooks': ['ipynb'],
     };
 
-    context.subscriptions.push(vscode.commands.registerCommand('polyglot-notebook.newNotebook', async () => {
+    async function newNotebookCommandHandler(useDefaults: boolean | undefined): Promise<void> {
+        const polyglotConfig = vscode.workspace.getConfiguration(constants.PolyglotConfigurationSectionName);
+        if (useDefaults) {
+            // try to get the default notebook type
+            const defaultNotebookExtension = polyglotConfig.get<string>('defaultNotebookExtension');
+            if (defaultNotebookExtension) {
+                await newNotebook(defaultNotebookExtension, true);
+                return;
+            }
+
+            // if we got here, there was no default set so we fall back to asking
+        }
+
         // offer to create either `.dib` or `.ipynb`
-        const newDibNotebookText = `Create as '.dib'`;
-        const newIpynbNotebookText = `Create as '.ipynb'`;
-        const selected = await vscode.window.showQuickPick([newDibNotebookText, newIpynbNotebookText]);
-        switch (selected) {
-            case newDibNotebookText:
-                await vscode.commands.executeCommand('polyglot-notebook.newNotebookDib');
-                break;
-            case newIpynbNotebookText:
-                await vscode.commands.executeCommand('polyglot-notebook.newNotebookIpynb');
-                break;
-            default:
-                break;
+        const availableNotebookExtensions = ['.dib', '.ipynb'];
+        const selectedExtension = await vscode.window.showQuickPick(availableNotebookExtensions, { title: 'Create as...' });
+        if (!selectedExtension) {
+            return;
         }
-    }));
 
-    context.subscriptions.push(vscode.commands.registerCommand('polyglot-notebook.newNotebookDib', async () => {
-        await newNotebook('.dib');
-    }));
+        if (useDefaults) {
+            // If we got here, the user entered via a path that expected to use a default, but it wasn't present.  Show
+            // an info box letting them know that a default can be specified.  The result isn't awaited because they'll
+            // likely want to continue with their current action.
+            promptToSaveConfigValue(`Would you like all future new notebooks to be created with the '${selectedExtension}' extension?`, selectedExtension, constants.PolyglotConfigurationSectionName, 'defaultNotebookExtension');
```

Comment:
```suggestion
            promptToSaveConfigValue(`Would you like future new notebooks to be created with the ${selectedExtension} extension?`, selectedExtension, constants.PolyglotConfigurationSectionName, 'defaultNotebookExtension');
```

Reply:
If we like the one above, following the format of the suggestion above, "Would you like future notebooks to use the *file extension* by default?"